controllers/users.js

```diff
@@ -113,6 +114,41 @@ const getUsers = async (req, res) => {
       });
     }
 
+    const profile = req.query.profile === "true";
+
+    if (profile) {
+      if (dev) {
+        let result, user;
+        let token = req.cookies[config.get("userToken.cookieName")];
+
+        // Extract token from authorization header if not in production and no token in cookies
+        if (process.env.NODE_ENV !== "production" && !token) {
+          token = req.headers.authorization ? req.headers.authorization.split(" ")[1] : null;
+        }
+
+        // Verify the token
+        let userId;
+        try {
+          ({ userId } = authService.verifyAuthToken(token));
+        } catch (error) {
+          logger.error(`Token verification failed: ${error}`);
+          return res.boom.unauthorized("Unauthenticated User");
+        }
```

Comment:
- should we move this logic inside a middleware that will be called on the `/users` route and pass the userId forward once the token has been verified?